lib/package.js

```diff
@@ -471,9 +471,20 @@ exports.download = function(pkg, options, installDeps) {
 
         // get the override
         if (endpoint.getOverride) {
-          if (!override && config.pjson.overrides[pkg.exactName])
-            ui.log('warn', 'Using local override for `' + pkg.exactName + '`');
-          return endpoint.getOverride(pkg.registry, pkg.package, pkg.version, override || config.pjson.overrides[pkg.exactName]);
+          if (!override) {
+            for (var overrideName in config.pjson.overrides) {
+              if (overrideName.startsWith(pkg.name)) {
```

Comment:
A package name might be contained within another package name - eg `github:components/jquery` and `github:components/jq` say. So we should check that `overrideName.startsWith(pkg.name + '@')`.